plugins/additional/additional.cpp

```diff
@@ -2480,6 +2481,109 @@ class RibbonGalleryItemComponent : public ComponentBase
 {
 };
 
+namespace
+{
+    class wxVListBoxPreview : public wxVListBox
+    {
+    public:
+        /* N.B.:  forbid one-step creation because the Create()
+            call is made within wxVListBox's ctor BEFORE
+            wxVListBoxPreview's ctor sets the proper
+            vtable */
+        wxVListBoxPreview() = default;
```

Comment:
I have realized that there is another solution to the problem that would enable both one-step and two-step creation, at the cost of a small amount of additional code:  If a one-step constructor is declared in this class, it will allow the `vtable` to be set before `Create()` is called.  Do you have a preference between the current, shorter code, or the alternative longer code that supports one-step creation?

Reply:
I think the root cause of the no-items issue is that the event gets processed when still inside the constructor. I think this can be solved by using `CallAfter()` instead. If you define the default constructor like this

```cpp
wxVListBoxPreview() { CallAfter([this] { SetItemCount(20); }); }
```

it should work, no need for a event table. The parameterized constructor could call the default constructor as delegate so you don't need to add that call there as well. Also please use a smaller item count than maximum integer, there is just no scrollbar thumb visible in that case.

Reply:
Done.  See https://github.com/wxFormBuilder/wxFormBuilder/pull/858/commits/1c08a58e05a52629c18d66e1678285cdf8177814.

However, I think you should report the no-visible-thumb problem to wxWidgets as a bug.  On my system (wxMSW, Windows 10 22H2), the thumb shows at the normal size with max_int items.

Reply:
I only used the wxFB build from GitHub on my Windows 11 machine, i just did another test with a regular wxListBox, there the scrollbar thumb is also missing, even when there are only a few items in it. Seems to be a generic problem specific to the Windows 11 "smart" scroll bars. After some more testing i figured out this happens when the scrollbar gets too small. The default height of the preview item is 50 pixels, at 60 pixels i get the thumb, even for your max int count of items.

![wxfb-scrollbars](https://github.com/user-attachments/assets/e661da50-4ede-45eb-a2c2-8502f7149bc8)